### PR TITLE
trivial: Do not rely on @memlock not being part of @system-service

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@RZR7332 do you know why this might be required? the daemon is able to do `mlock()` even though we're allow-listing just `@system-service`

**Comment by RZR7332** [comment]
> @RZR7332 do you know why this might be required? the daemon is able to do `mlock()` even though we're allow-listing just `@system-service`

Thanks for picking this up, suspect it might be nested under either system-service or privileged. Let me dig a bit and will come back to you either this morning or at latest some time tomorrow.

**Comment by RZR7332** [comment]
It looks like I have goofed (but have now also learned something new, so thank you!) - it is nested under system-service (output omitted for brevity):

@system-service
    # General system service operations
    @aio
    @basic-io
    @chown
    @default
    @file-system
    @io-event
    @ipc
    @keyring
    @memlock
    @network-io
    @process
    @resources
    @setuid
    @signal
    @sync
    @timer

Note @privileged is not listed here, even though it is included. That group contains the below (again, output omitted for brevity):

@privileged
    # All system calls which need super-user capabilities
    @chown
    @clock
    @module
    @raw-io
    @reboot
    @swap

This means that @raw-io will always be included ultimately as part of system-service, so perhaps the dynamic options can be changed rather to blacklist it if flashrom is not present, rather than the current whitelist?

My suggestion would be to leave @system-service as the whitelisted option, and then to blacklist the calls which were previously dropped before my PR changed it to whitelist. It is obviously possible to be incredibly granular about which specific calls are allowed, but that would be a lot of work and not details I possess. What do you think?

Evidently my reliance on systemd-analyze security is a big part of the issue here -_- my apologies. I shall update my own documentation and service files accordingly.

**Comment by hughsie** [comment]
@RZR7332 `systemd-analyze syscall-filter` is super useful here. I've pushed a change to add an explicit allow-list, not using `@system-service` as that's just too broad in scope.

**Comment by RZR7332** [comment]
@hughsie that is the command I have been using all along, agree very useful with providing details. My poor reading/inspection aside - your idea is far better. Thank you for taking the time to look into this and improve the filtering!

File: data/meson.build [hunk 1]
```diff
@@ -94,15 +94,22 @@ if build_daemon
...
+      '@ipc',
+      '@network-io',
+      '@process',
+      '@sync',
+      'ioctl',
+      'uname',
+    ]
     if allow_flashrom
-      syscall_allow += ['@raw-io']
+      syscall_filter += ['@raw-io']
```
Comment (superm1) [inline data/meson.build:110]: Shouldn't raw io NOT be filtered when allow flashrom?

Comment (hughsie) [reply data/meson.build:110]: It's an allow-list only with this PR.

Comment (superm1) [reply data/meson.build:110]: How counterintuitive how this stuff works. Ok.

**Review by superm1 — APPROVED** [review]

**Pull request merged**